graphql_compiler/schema_transformation/split_query.py

```diff
@@ -34,6 +46,392 @@ def __init__(self, query_ast):
+            else:
+                intra_schema_fields.append(vertex_field)
+        elif isinstance(vertex_field, InlineFragment):
+            intra_schema_fields.append(vertex_field)
+        else:
+            raise AssertionError(
+                u'Input vertex field {} is neither a Field nor an InlineFragment'.format(
+                    vertex_field
+                )
+            )
+    return intra_schema_fields, cross_schema_fields
+
+
+def _get_selections_from_property_and_vertex_fields(property_fields_map, vertex_fields):
+    """Combine property fields and vertex fields into a list of selections.
+
+    Args:
+        property_fields_map: OrderedDict[str, Field], mapping name of field to their
+                             representation. It is not modified by this function
+        vertex_fields: List[Union[Field, InlineFragment]]. It is not modified by this function
+
+    Returns:
+        List[Union[Field, InlineFragment]], containing all property fields then all vertex fields,
+        in order
+    """
+    selections = list(six.itervalues(property_fields_map))
+    selections.extend(vertex_fields)
+    return selections
+
+
+def _get_child_query_node_and_out_name(ast, child_type_name, child_field_name,
+                                       intermediate_out_name_assigner):
+    """Create a query node out of ast, return node and unique out_name on field with input name.
+
+    Create a new document out of the input AST, that has the same structure as the input. For
+    instance, if the input AST can be represented by
+        out_Human {
+          name
+        }
+    where out_Human is a vertex field going to type Human, the resulting document will be
+        {
+          Human {
+            name
+          }
+        }
+    If the input AST starts with a type coercion, the resulting document will start with the
+    coerced type, rather than the original union or interface type.
+
+    The output child_node will be wrapped around this new Document. In addition, if no field
+    of child_field_name currently exists, such a field will be added. If there is no @output
+    directive on this field, a new @output directive will be added.
+
+    Args:
+        ast: type Node, representing the AST that we're using to build a child node. It is not
+             modified by this function
+        child_field_name: str. If no field of this name currently exists as a part of the root
+                          selections of the input AST, a new field will be created in the AST
+                          contained in the output child query node
+        intermediate_out_name_assigner: IntermediateOutNameAssigner, which will be used to
+                                        generate an out_name, if it's necessary to create a
+                                        new @output directive
+
+    Returns:
+        Tuple[SubQueryNode, str], the child sub query node wrapping around the input ast, and
```

Comment:
nit: `ast` -> `AST` since "input ast" doesn't seem to be referring to a function parameter